### PR TITLE
Feature/emberizing

### PR DESCRIPTION
**Pull request by mehran-naghizadeh**
This PR seeks to implement the **leidinggevendendatabank** frontend app.

Todo:
1. Take careof the _LEIDINGGEVENDEN_ link on the top of page.
2. Take care of the _CONTACT ONS_ link.
3. Take care of links in the _Download leidinggevendendatabank_ section.
4. Find out what the purpose of _templates/components/head.hbs_ is.

### CODE REVIEW

**Comment by cecemel** [comment]
Nice!

some functional remarks first

1. it was unspecified, but we should see only the current functionarissen, so filter on these with end in future or unspecified

2. take e.g the url: bestuurseenheid/974816591f269bb7d74aa1720922651529f3d3b2a787f5c60b73e5a0384950a4/functionarissen
 -> don't we need like email, website, phone fax?

3. when selecting a bestuurseenheid, going to the table, and then going back, it would be nice to keep the selectbox filled in with the selected bestuurseenheid 
look in the example: https://mandaten.lokaalbestuur.vlaanderen.be -> look both a the functionality, the url params and the code. 

4. Rdfa annotations are stil missing, but you are busy with this I suppose :+1: 

5. there is a problem indeed with fastboot and emberdata table, let's have a look

File: app/components/select-bestuurseenheid.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { task, timeout } from 'ember-concurrency';
+
+export default Component.extend({
+  store: service(),
+
+  async init() {
+    this._super(...arguments);
+    const options = this.store.query('bestuurseenheid', {
+      sort: 'naam',
+      include: ['classificatie']
```
Comment (cecemel) [inline app/components/select-bestuurseenheid.js]: This translates to wrong url, it should be 
```
{
  include: 'relation-1,relation-2'
}

Comment (cecemel) [reply app/components/select-bestuurseenheid.js]: it translates now to this: http://localhost:4200/bestuurseenheden?include[]=classificatie&sort=naam

File: app/components/select-bestuurseenheid.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+
+export default Component.extend({
+  store: service(),
+
+  async init() {
+    this._super(...arguments);
+    const options = this.store.query('bestuurseenheid', {
```
Comment (cecemel) [inline app/components/select-bestuurseenheid.js:10]: I need to be convinced of the use of loading the data on init. It slows a little. 
We don't do it on the mandatendatabank either.

Comment (cecemel) [reply app/components/select-bestuurseenheid.js:10]: see https://mandaten.lokaalbestuur.vlaanderen.be/

File: app/components/select-bestuurseenheid.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      sort: 'naam',
+      include: ['classificatie']
+    });
+    this.set('options', options);
+  },
+
+  async didReceiveAttrs() {
+    this._super(...arguments);
+    if (this.value && !this.selected) {
```
Comment (cecemel) [inline app/components/select-bestuurseenheid.js:19]: what is this.value ?

File: app/components/select-bestuurseenheid.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      const bestuurseenheids = this.store.query('bestuurseenheid', {
+        filter: { id: this.value },
```
Comment (cecemel) [inline app/components/select-bestuurseenheid.js]: if this.value is an id, woudn't .findRecord be better?

**Comment by mehran-naghizadeh** [comment]
```
-> don't we need like email, website, phone fax?
```

Why do we need them? They are not included in the table for except the email.

**Review by cecemel — COMMENTED** [review]
> ```
> -> don't we need like email, website, phone fax?
> ```
> 
> Why do we need them? They are not included in the table for except the email.

The phone number is here: https://github.com/lblod/frontend-leidinggevenden-databank/pull/1/commits/dd8068c4dbf064068583647331baf49abc51849b#diff-34aa55193f9ba6a117b2473e04339b3eL80
I would include all the stuff we have so far, and let the designer figure out what would be a better way of presenting it.

File: app/components/select-bestuurseenheid.js (hunk 4 above)
Comment (mehran-naghizadeh) [reply app/components/select-bestuurseenheid.js]: It is already a _findRecord_ call. Look [here](https://github.com/lblod/frontend-leidinggevenden-databank/blob/05ec1c394d2760d5b080023bb0c650eafa184f09/app/components/select-bestuurseenheid.js#L20).

File: app/components/select-bestuurseenheid.js (hunk 3 above)
Comment (mehran-naghizadeh) [reply app/components/select-bestuurseenheid.js:19]: The _bestuurseenheid id_  included in the page url.

File: app/components/select-bestuurseenheid.js (hunk 2 above)
Comment (mehran-naghizadeh) [reply app/components/select-bestuurseenheid.js:10]: I believe this should be compared to the [bestuurseenheid-select](https://github.com/lblod/frontend-toezicht-abb/blob/c236a20447f353f0c4815b74fd353aa86a6d12d8/app/components/bestuurseenheid-select.js#L10) component from the frontend-toezicht-abb app.

File: app/routes/bestuurseenheid/functionarissen.js [hunk 5]
```diff
@@ -9,7 +9,8 @@ export default Route.extend(DataTableRouteMixin, {
   mergeQueryOptions(params) {
     this.set('bestuurseenheidId', params.bestuureenheid_id);
     return {
-      'filter[bekleedt][bevat-in][is-tijdsspecialisatie-van][bestuurseenheid][:id:]': params.bestuureenheid_id
+      'filter[bekleedt][bevat-in][is-tijdsspecialisatie-van][bestuurseenheid][:id:]': params.bestuureenheid_id,
+      'filter[:gte:einde]': new Date().toISOString()
```
Comment (cecemel) [inline app/routes/bestuurseenheid/functionarissen.js]: This won't work for einde: undefined.
Unfortunately, you can not ask mu-cl-resources for undefined values, so you will have to come up with a slightly more complex filter. 
Also, best to test your filter too. Here is an example call, 

```
/functionarissen?filter[bekleedt][bevat-in][is-tijdsspecialisatie-van][bestuurseenheid][:id:]=974816591f269bb7d74aa1720922651529f3d3b2a787f5c60b73e5a0384950a4&sort=is-bestuurlijke-alias-van.achternaam
```

**Comment by cecemel** [comment]
already in dev

**Pull request closed without merging**